src/menu/Menu.ts

```diff
@@ -0,0 +1,299 @@
+import MenuItem from './MenuItem';
+import * as css from './styles/menu.m.css';
+
+export type Role = 'menu' | 'menubar';
+
+/**
+ * @type MenuProperties
+ *
+ * Properties that can be set on a Menu component
+ *
+ * @property animate
+ * Only applies to nested menus. A flag indicating whether the widget instance should handle animating between the
+ * visible and hidden states. If true (the default), then the menu will slide into and out of view like an accordion.
+ * If false, then any animation should be handled in the CSS, as the menu will just snap open/shut.
+ *
+ * @property disabled
+ * Indicates whether the menu is disabled. If true, then the widget will ignore events.
+ *
+ * @property expandOnClick
+ * Indicates whether the menu should be displayed/hidden via a click event. If false, then the menu is toggled
+ * via a hover event. Defaults to true.
+ *
+ * @property hideDelay
+ * Only applies to menus toggled into view with a hover event. The amount of time (in milliseconds) after a
+ * mouseleave event that should pass before the menu is hidden. Defaults to 300ms.
+ *
+ * @property hidden
+ * Whether the menu is hidden. Defaults to true if a label is specified (i.e., the menu is controlled by a
+ * link); false otherwise.
+ *
+ * @property id
+ * The widget ID. Defaults to a random string.
+ *
+ * @property label
+ * A widget that will be wrapped in a MenuItem widget that will be used to control the menu.
+ *
+ * @property nested
+ * Indicates whether the menu is nested within another menu. Useful for styling, this does not affect
+ * functionality. Defaults to false.
+ *
+ * @property onRequestHide
+ * Needed only when a label is used. Called when the menu is displayed, and the label is triggered.
+ * This method should be used to update the widget's `hidden` property.
+ *
+ * @property onRequestShow
+ * Needed only when a label is used. Called when the menu is hidden, and the label is triggered.
+ * This method should be used to update the widget's `hidden` property.
+ *
+ * @property role
+ * The value to use for the menu's `role` property. Defaults to "menu".
+ */
+export interface MenuProperties extends ThemeableProperties {
+	animate?: boolean;
+	disabled?: boolean;
+	expandOnClick?: boolean;
+	hideDelay?: number;
+	hidden?: boolean;
+	id?: string;
+	label?: DNode;
+	nested?: boolean;
+	onRequestHide?: () => void;
+	onRequestShow?: () => void;
+	role?: Role;
+}
+
+function getMenuHeight(menuElement: HTMLElement): number {
+	const maxHeight = parseInt(getComputedStyle(menuElement).getPropertyValue('max-height'), 10);
+	return Math.min(menuElement.scrollHeight, (isNaN(maxHeight) ? Infinity : maxHeight));
+}
+
+export const MenuBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export class Menu extends MenuBase<MenuProperties> {
+	protected wasOpen = false;
+	private _hideTimer: Handle;
+	private _initialRender = true;
+
+	render(): DNode {
+		const {
+			id = uuid(),
+			nested,
+			role = 'menu'
```

Comment:
True. It should probably be tied to whatever determines whether or not we use `nav`, if we decide to control that.